addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/NestBindingConstants.java

```diff
@@ -98,9 +102,8 @@
     public static final String CHANNEL_TIME_ZONE = "time_zone";
     public static final String CHANNEL_ETA_BEGIN = "eta_begin";
     public static final String CHANNEL_RUSH_HOUR_REWARDS_ENROLLMENT = "rush_hour_rewards_enrollment";
-    public static final String CHANNEL_AWAY = "away";
 
     // Properties on things that are fixed.
-    public static final String PROPERTY_ID = "deviceId";
-    public static final String PROPERTY_FIRMWARE_VERSION = "firmware";
+    public static final @NonNull String PROPERTY_ID = "deviceId";
+    public static final @NonNull String PROPERTY_FIRMWARE_VERSION = "firmware";
```

Comment:
Is this the expected way to go? Why don't we use the static finals `PROPERTY_MODEL_ID` and `PROPERTY_FIRMWARE_VERSION` defined in the [ESH Thing.java](https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/Thing.java#L32)?

Should the `@NonNull` annotation be added over there?

Reply:
Thanks, will use them instead! Don't know if I will still get the \@NonNull warning in that case.

Reply:
If the `@NonNull` warning is caused by the `updateProperty()` method you will still see it.

Reply:
Pitty:

```
WARNING] /Users/mvanwinger/private/openhab2-addons/addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestThermostatHandler.java:[123] 
	updateProperty(PROPERTY_MODEL_ID, thermostat.getDeviceId());
	               ^^^^^^^^^^^^^^^^^
Null type safety (type annotations): The expression of type 'String' needs unchecked conversion to conform to '@NonNull String'
[WARNING] /Users/mvanwinger/private/openhab2-addons/addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestThermostatHandler.java:[124] 
	updateProperty(PROPERTY_FIRMWARE_VERSION, thermostat.getSoftwareVersion());
	               ^^^^^^^^^^^^^^^^^^^^^^^^^
Null type safety (type annotations): The expression of type 'String' needs unchecked conversion to conform to '@NonNull String'
[WARNING] /Users/mvanwinger/private/openhab2-addons/addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestStructureHandler.java:[94] 
	updateProperty(PROPERTY_MODEL_ID, structure.getStructureId());
	               ^^^^^^^^^^^^^^^^^
Null type safety (type annotations): The expression of type 'String' needs unchecked conversion to conform to '@NonNull String'
[WARNING] /Users/mvanwinger/private/openhab2-addons/addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestCameraHandler.java:[73] 
	updateProperty(PROPERTY_MODEL_ID, camera.getDeviceId());
	               ^^^^^^^^^^^^^^^^^
Null type safety (type annotations): The expression of type 'String' needs unchecked conversion to conform to '@NonNull String'
[WARNING] /Users/mvanwinger/private/openhab2-addons/addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestCameraHandler.java:[74] 
	updateProperty(PROPERTY_FIRMWARE_VERSION, camera.getSoftwareVersion());
	               ^^^^^^^^^^^^^^^^^^^^^^^^^
Null type safety (type annotations): The expression of type 'String' needs unchecked conversion to conform to '@NonNull String'
[WARNING] /Users/mvanwinger/private/openhab2-addons/addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestSmokeDetectorHandler.java:[73] 
	updateProperty(PROPERTY_MODEL_ID, smokeDetector.getDeviceId());
	               ^^^^^^^^^^^^^^^^^
Null type safety (type annotations): The expression of type 'String' needs unchecked conversion to conform to '@NonNull String'
[WARNING] /Users/mvanwinger/private/openhab2-addons/addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestSmokeDetectorHandler.java:[74] 
	updateProperty(PROPERTY_FIRMWARE_VERSION, smokeDetector.getSoftwareVersion());
	               ^^^^^^^^^^^^^^^^^^^^^^^^^
Null type safety (type annotations): The expression of type 'String' needs unchecked conversion to conform to '@NonNull String'
7 problems (7 warnings)
```